src/Discovery/DiscoveryActionBar.tsx

```diff
@@ -31,23 +34,63 @@ interface Props {
   user: User,
   downloadInProgress: boolean,
   setDownloadInProgress: (boolean) => void,
-  downloadStatusMessage: {
+  discoveryActionStatusMessage: {
     title: string,
     message: string,
     active: boolean,
     url: string
   },
-  setDownloadStatusMessage: (Object) => void;
+  setDiscoveryActionStatusMessage: (Object) => void;
 }
 
+const checkFederatedLoginStatus = async (
+  setDiscoveryActionStatusMessage: (arg0: Object) => void,
+) => fetchWithCreds({
+  path: `${externalLoginOptionsUrl}`,
+  method: 'GET',
+}).then(
+  ({ data, status }) => {
+    if (status !== 200) {
+      return false;
+    }
+
+    const { providers } = data;
+    const unauthenticatedProviders = providers.filter(
+      (provider) => !provider.refresh_token_expiration,
```

Comment:
can't remember what is the logic on WTS side, but if user has logged in before and refresh token has expired, will the `refresh_token_expiration` presists?

Reply:
I didn't think to check this before, but if a user's refresh token is expired in wts, wts will return `none` for that idp
https://github.com/uc-cdis/workspace-token-service/blob/3f58479627b2af9b2d1210b40e2126dc1c359acb/wts/blueprints/external_oidc.py#L136